### PR TITLE
Add check_requires() function and update installation docs

### LINKED ISSUE #46: Add suggested dependencies to setup documentation
Setup documentation should include suggested R packages and what they are required for.

### LINKED ISSUE #47: Add checks for suggested packages
Anything that needs a suggested package should check if it's available and give a nice message about installing it if not

### PR DESCRIPTION
**Pull request by lazappi**
Added a function that checks if suggested packages are installed and gives a nice error with installation instructions if not. 

Updated installation docs with instructions for installing suggested dependencies and information on what they are used for.

Fixes #46 
Fixes #47

### CODE REVIEW

**Review by rcannood — CHANGES_REQUESTED** [review]
Nice, great work!

* Could you add an entry to the changelog?
* Could you add a small unit test to check that `check_requires("foobar")` throws an error but `check_requires("laminr")` doesn't?

File: tests/testthat/test-utils.R [hunk 1]
```diff
@@ -0,0 +1,7 @@
+test_that("check_requires works", {
+  expect_true(check_requires("Imported packages", "cli"))
+  expect_error(
+    check_requires("Missing packages", "a_missing_package"),
+    regexp = "Missing packages requires"
+  )
+})
```
Comment (lazappi) [inline tests/testthat/test-utils.R:7]: Updated the CHANGELOG. There are already these tests unless you want to check specific packages?

Comment (rcannood) [reply tests/testthat/test-utils.R:7]: I overlooked on the first review, apologies!

**Comment by rcannood** [comment]
LGTM!

**Review by rcannood — APPROVED** [review]

**Pull request merged**